azure/scope/managedmachinepool.go

```diff
@@ -224,6 +224,8 @@ func buildAgentPoolSpec(managedControlPlane *infrav1.AzureManagedControlPlane,
 		}
 	}
 
+	agentPoolSpec.LinuxOSConfig = managedMachinePool.Spec.LinuxOSConfig
```

Comment:
any reason for not setting it above when the struct is initialized https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2973/files#diff-5607bc7afbe8f18686e0e62796b8da6371e999006ddb598c5034dd7ef7040a47R157 ?

Reply:
No specific reason apart from signifying that it is a recent addition. But I think git blame or history would point that out too. Moving it. :)